contracts/verifiers/ZKPVerifier.sol

```diff
@@ -4,18 +4,38 @@ pragma solidity 0.8.16;
 import {ERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
 import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
 import {GenesisUtils} from "../lib/GenesisUtils.sol";
-import {PoseidonFacade} from "../lib/Poseidon.sol";
 import {ICircuitValidator} from "../interfaces/ICircuitValidator.sol";
 import {IZKPVerifier} from "../interfaces/IZKPVerifier.sol";
+import {ArrayUtils} from "../lib/ArrayUtils.sol";
 
 contract ZKPVerifier is IZKPVerifier, Ownable {
```

Comment:
Can ever ZKPVerifier be deployed directly? Should it be abstract if "No"?

Reply:
I think yes. ERC20 is not abstract.

Reply:
Yes, it can.